server/sonar-server/src/main/java/org/sonar/server/computation/step/PersistComponentsStep.java

```diff
@@ -67,63 +66,58 @@ public void execute() {
     DbSession session = dbClient.openSession(false);
     try {
       org.sonar.server.computation.component.Component root = treeRootHolder.getRoot();
-      List<ComponentDto> components = dbClient.componentDao().selectComponentsFromProjectKey(session, root.getKey());
-      Map<String, ComponentDto> componentDtosByKey = componentDtosByKey(components);
-      PersisComponentExecutor componentContext = new PersisComponentExecutor(session, componentDtosByKey, reportReader, reportReader.readMetadata().getAnalysisDate());
+      List<ComponentDto> existingComponents = dbClient.componentDao().selectComponentsFromProjectKey(session, root.getKey());
+      Map<String, ComponentDto> existingComponentDtosByKey = componentDtosByKey(existingComponents);
+      PersisComponent persisComponent = new PersisComponent(session, existingComponentDtosByKey, reportReader);
 
-      componentContext.recursivelyProcessComponent(root, null, null);
+      persisComponent.recursivelyProcessComponent(root, null);
       session.commit();
     } finally {
       session.close();
     }
   }
 
-  private class PersisComponentExecutor {
+  private class PersisComponent {
```

Comment:
I don't understand what this class does from its name
the previous name gave a little more information